commons-db/src/main/java/com/palantir/nexus/db/sql/SQLString.java

```diff
@@ -251,22 +252,40 @@ private static String makeCommentString(String keyString, DBType dbType) {
     /**
      * Cleans up whitespace, any trailing semicolon, and prefixed comments that a string is
      * unregistered, in order to come up with a canonical representation of this sql string.
-     * @param s
-     * @return
      */
-    public static String canonicalizeString(String s) {
-        String trim = s.trim();
-        if(!trim.isEmpty() && trim.charAt(trim.length() - 1) == ';') {
-            trim = trim.substring(0, trim.length() - 1);
+    public static String canonicalizeString(String original) {
```

Comment:
looks like the only caller of this method in AtlasDB immediately does a copy of the returned String and strips out all whitespace, so might be worthwhile to create a private version that does not copy over any whitespace. Note that there is at least one caller of `canonicalizeString` outside of this project so we'd still need the public method to return a String

```
        FinalSQLString cached = cachedUnregistered.get(StringUtils.deleteWhitespace(canonicalizeString(sql)));
```